File: 101/exercises/easy3/palindromic.rb

def palindrome?(reversible)
  reversible == reversible.reverse
end

def real_palindrome?(str)
  alphanum_part = str.downcase.gsub(/[^\w\d]/, '')
  # alternative: .delete('^a-z0-9')
  palindrome? alphanum_part
end

def palindromic_number?(num)
  palindrome? num.to_s
end

## Tests

p palindrome?('madam') == true
p palindrome?('Madam') == false          # (case matters)
p palindrome?("madam i'm adam") == false # (all characters matter)
p palindrome?('356653') == true
p palindrome?(%w(blah blubb blubb blah)) == true
p palindrome?([1, 2, 3, 4]) == false
p palindrome?('') == true
p palindrome?([]) == true

p real_palindrome?('madam') == true
p real_palindrome?('Madam') == true           # (case does not matter)
p real_palindrome?("Madam, I'm Adam") == true # (only alphanumerics matter)
p real_palindrome?('356653') == true
p real_palindrome?('356a653') == true
p real_palindrome?('123ab321') == false

p palindromic_number?(34543) == true
p palindromic_number?(123210) == false
p palindromic_number?(22) == true
p palindromic_number?(5) == true

File: 210/small-problems/easy3/palindromic.js

function isPalindrome(string) {
  return string === string.split('').reverse().join('');
}

function isRealPalindrome(string) {
  return isPalindrome(string.toLowerCase().replace(/[^\w\d]/g, ''));
}

function isPalindromicNumber(number) {
  return isPalindrome(String(number));
}

// Tests

console.log(isPalindrome('madam') === true);
console.log(isPalindrome('Madam') === false); // case matters
console.log(isPalindrome("madam i'm adam") === false); /// all characters matter
console.log(isPalindrome('356653') === true);

console.log(isRealPalindrome('madam') === true);
console.log(isRealPalindrome('Madam') === true); // case does not matter
console.log(isRealPalindrome("Madam, I'm Adam") === true); // only alphanumerics matter
console.log(isRealPalindrome('356653') === true);
console.log(isRealPalindrome('356a653') === true);
console.log(isRealPalindrome('123ab321') === false);

console.log(isPalindromicNumber(34543) === true);
console.log(isPalindromicNumber(123210) === false);
console.log(isPalindromicNumber(22) === true);
console.log(isPalindromicNumber(5) === true);
